core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -359,7 +359,13 @@ private boolean manifestHasDeletedFiles(PartitionAndMetricsEvaluator evaluator,
       boolean fileDelete = deletePaths.contains(file.path()) ||
           dropPartitions.contains(file.specId(), file.partition()) ||
           (isDelete && entry.sequenceNumber() > 0 && entry.sequenceNumber() < minSequenceNumber);
+
       if (fileDelete || evaluator.rowsMightMatch(file)) {
+        if (!fileDelete && !file.content().equals(FileContent.DATA) && !evaluator.rowsMustMatch(file)) {
```

Comment:
I think if we can I would un-nest this clause and assign it to a variable describing what we are checking here (much like fileDelete) above